tree/tree/src/TTree.cxx

```diff
@@ -639,6 +639,44 @@ Long64_t TTree::TClusterIterator::Next()
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// Move on to the previous cluster and return the starting entry
+/// of this previous cluster
+
+Long64_t TTree::TClusterIterator::Previous()
+{
+   fNextEntry = fStartEntry;
+   if ( fTree->GetAutoFlush() <= 0 ) {
+      // Case of old files before November 9 2009
+      Long64_t clusterEstimate = GetEstimatedClusterSize();
+      fStartEntry = fNextEntry - clusterEstimate;
+   } else {
+      if (fClusterRange == 0) {
```

Comment:
Shouldn't this be ``` == fTree->fNClusterRange ``` ?

Reply:
HI Philippe,

You are ~correct. Checking the CI build failure, I see that Previous() did not do the right thing for TTrees without cluster ranges. Apparently the fClusterRange == fTree->fNClusterRange is not only catching the last range, but also if fNClusterRange == 0, and in that case uses the auto-flush setting.

So the first condition on line 653 should be:
// catching the first range or trees without cluster ranges
if (fClusterRange == 0 || fTree->fNClusterRange == 0) {
And the second on line 661 remains:
// catching the first range 
if (fClusterRange == 0) {

I would almost argue that in the existing Next(), on line 609 is a misleading comment and should be
// We are looking at a range which size
// is defined by AutoFlush itself and goes to the GetEntries.

I made these changes and that cures the build failure (which was ~random). In addition, I made the Travis CI formatting changes, except those requested for the TTree.h file (whitespace, that makes the file look really inconsistent). I am also hesitant of changing TTree.cxx constructor to initialize multiple members on the same line, but can include that in the next commit (please let me know).

Thanks, Peter

Reply:
> I made the Travis CI formatting changes, ... I am also hesitant of changing 

Indeed I agree.  In first approximation trust the surrounding code more than clang-format :) ... at least for now.

Reply:
> I would almost argue that in the existing Next(), on line 609 is a misleading comment and should be

Indeed your version is clearer.